lib/kitchen/driver/aws/instance_generator.rb

```diff
@@ -62,22 +62,38 @@ def ec2_instance_data # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
           end
 
           if config[:security_group_ids].nil? && config[:security_group_filter]
-            security_group = ::Aws::EC2::Client
-                .new(region: config[:region]).describe_security_groups(
-                filters: [
-                    {
-                        name: "tag:#{config[:security_group_filter][:tag]}",
-                        values: [config[:security_group_filter][:value]],
-                    },
+            security_groups = []
+            filters = [config[:security_group_filter]].flatten
+            filters.each do |sg_filter|
+              r = {}
+              if sg_filter[:name]
+                r[:filters] = [
+                  {
+                    name: "group-name",
```

Comment:
What would happen if we have multiple `name` entries? 🤔 Something like:
```
security_group_filter:
  - name: 'AWS-A'
  - name: 'AWS-B'
  - tag: 'Name'
    value: 'my-app'
```
Is it allowed to have multiple `group-name` entries or should we restrict/check to have only one inside `filters`?

Reply:
@afiune multiple group name entries are fine.  In the case there it would grab all 3 security groups.

Reply:
I think the better answer to your question is that it's one search per array entry.  So yeah youd be searching 3 times and being dropped into all 3 security groups.  This is the intended behavior.

Would love to see this merged into ChefDK/Workstation prior to the next release-- I've already poked my Chef rep about it in June.

Reply:
Makes perfect sense! Thanks for that answer. We are looking into merging this PR 💯